### PR TITLE
Enhancements for PouchORM

### PR DESCRIPTION
**Pull request by aaronhuggins**
Hi! I'm going to use PouchORM in a production app in the next month or two, and wow. What a great project. Let me tell you, I'm (mostly) thrilled with where it is at today.

There were some must haves for ORM that might have held me back, but this was just too good a Library to pass up. So here is a pull request for non-breaking enhancements. I've added some stuff that's just syntactic sugar depending on how you look at it, but I think it was all justified:

- Delta function as an optional parameter on `upsert`
- Object classing to support `new Model({ SOME_OBJECT })` syntax and runtime class validation
- Class property validation using the great `class-validator` library as an optional dependency

Let me know what you think. I hope these are in-line with your vision for where this project is headed. I'm super excited for you whether you accept this PR or not, because PouchORM as it stands today is a phenomenal start to bringing ORM to PouchDB.

### CODE REVIEW

**Comment by iyobo** [comment]
@aaronhuggins Awesome feature list!
Class based models for runtime validation is something I specifically thought of adding while relaxing during the holidays.

I'll have a look at this PR in more detail over the weekend.
Thanks!

File: src/tests/all.test.ts [hunk 1]
```diff
@@ -198,7 +209,36 @@ describe('PouchCollection Instance', () => {
                 expect(newguys).toHaveLength(0);
             });
 
-        })
+        });
+        describe('upsert with', () => {
```
Comment (iyobo) [inline src/tests/all.test.ts]: Great tests! 
But could you also add some quick ones under "with data" to cover finding items from a PouchCollection created by class models?

Just `find` should be fine.

Comment (aaronhuggins) [reply src/tests/all.test.ts]: Can do!

Comment (aaronhuggins) [reply src/tests/all.test.ts]: This should be fixed.

**Comment by iyobo** [comment]
pouch-orm@1.2.0 deployed to NPM 👍

**Comment by aaronhuggins** [comment]
> pouch-orm@1.2.0 deployed to NPM 👍

Awesome, @iyobo! Thanks!

Already working on an collection 'plugin' for auth using [pouchdb-auth](https://github.com/pouchdb/pouchdb-server/tree/master/packages/node_modules/pouchdb-auth) as a separate library over at [pouchorm-auth](https://github.com/aaronhuggins/pouchorm-auth). I'm using a version of this already in my nestjs app, and decided to modularize the code. Thought you might be interested in case you ever consider doing a plugin architecture for PouchORM, especially since this is early stages (only been a month, right?).

**Pull request merged**